Automattic/pocket-casts-ios
# Bookmarks: Adds a way to display Toast messages

| 🎨 Designs: [Figma](https://www.figma.com/file/Io7PCz1H1hmOgEE9eljoYW/Bookmarks?type=design&node-id=2032-52595&mode=dev) |
|:---:|

This adds a way to display a temporary notification overlay message with actions. The view can be dismissed either automatically after a timeout period, or by flicking the message up or down.

The Toast class is designed to be global so it uses a singleton to keep track of the visible toast window and dismiss it when needed. 

Simple notifications can be displayed by using 

```swift
Toast.show("Hello World")
```

## Demo Video

https://github.com/Automattic/pocket-casts-ios/assets/793774/fc492f9c-1aa8-45ac-ac5a-f295311d48d1

## To test

1. Enable the Bookmarks Feature Flag by going to Profile > Cog > Beta Features > bookmarks
2. Open the Full Screen Player
3. Tap on the Add Bookmark player shelf item
   - This has been replaced with a debug action
4. ✅ Verify you see the toast message
5. Wait 5 seconds
6. ✅ Verify the view is automatically hidden
7. Tap the add bookmark action again
8. Swipe up or down to dismiss the action
9. ✅ Verify it fades out and dismisses in the direction you flicked
10. Tap the add bookmark action again
11. Tap the 'View' button in the action
12. ✅ Verify you see 'View Action` printed in console and the toast is hidden automatically

## Checklist
